### PR TITLE
[#10] [Flutter] [Chore] Implement a cross env configuration for web & mobile

### LINKED ISSUE #10: [Flutter] [Chore] Implement a cross env configuration for web & mobile
## Why

Regarding this task https://github.com/nimblehq/flutter_mqtt_showcase/issues/9, we need to find a better way to support env reading from a `.env` file for all possible platforms (web + mobile and maybe also desktop). 

Mostly recheck [this PR](https://github.com/nimblehq/showcase_flutter_build_ios_android/pull/13) and [the previous implementation](https://github.com/nimblehq/showcase-flutter-build-web-desktop/pull/41) why do we need to switch back to flutter_config?

## Who Benefits?

Developers

### PR DESCRIPTION
**Pull request by hoangnguyen92dn**
- Close #10

## What happened 👀

Cherry-pick update from Pooh: https://github.com/nimblehq/showcase-flutter-build-web-desktop/pull/41

## Insight 📝

Bring the updates from Pooh to this repo

Cherry-pick step: https://gist.github.com/bhumphrey/3764983?permalink_comment_id=4601290#gistcomment-4601290

## Proof Of Work 📹

N/A

### CODE REVIEW

**Comment by luongvo** [comment]
@hoangnguyen92dn please try to totally remove flutter_config on both Android and iOS 🙏 

```
Launching lib/main.dart on Pixel 2 XL in debug mode...
Running Gradle task 'assembleDebug'...                          

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/luongvo/WORKS/_nimblehq/_flutter/flutter_mqtt_showcase/android/app/build.gradle' line: 32

* What went wrong:
A problem occurred evaluating project ':app'.
> Project with path ':flutter_config' could not be found in project ':app'.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 16s
Running Gradle task 'assembleDebug'...                             18.9s
```

File: pubspec.yaml [hunk 1]
```diff
@@ -47,9 +47,11 @@ dependencies:
   retrofit: ^4.0.1
   equatable: ^2.0.5
   rxdart: ^0.27.7
+  hive: ^2.2.3
+  hive_flutter: ^1.1.0
```
Comment (luongvo) [inline pubspec.yaml:51]: Why do we need to use hive? Wrong cherry-picking, right?

**Review by luongvo — CHANGES_REQUESTED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (hoangnguyen92dn) [reply pubspec.yaml:51]: Hmm...The dual display team doesn't use hive and they disabled the dependencies 😮‍💨

File: pubspec.lock [hunk 2]
```diff
@@ -453,6 +453,22 @@ packages:
...
+    source: hosted
+    version: "2.2.3"
+  hive_flutter:
+    dependency: "direct main"
+    description:
+      name: hive_flutter
+      sha256: dca1da446b1d808a51689fb5d0c6c9510c0a2ba01e22805d492c73b68e33eecc
+      url: "https://pub.dev"
+    source: hosted
+    version: "1.1.0"
```
Comment (luongvo) [inline pubspec.lock:471]: please refresh .lock file

**Review by luongvo — APPROVED** [review]

**Pull request merged**